File: Data Structures & Algorithms/sliding-window-maximum/submission-2.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        Deque<Integer> deque = new ArrayDeque();
        int[] res = new int[nums.length - k + 1];
        // check if index is at front of the deque is now outside <= i-k
        for (int i = 0; i < nums.length; i++) {
            if (!deque.isEmpty() && deque.peekFirst() <= i - k) {
                deque.pollFirst();
            }
            // evict from the back while it is smaller
            while (!deque.isEmpty() && nums[deque.peekLast()] < nums[i]) {
                deque.pollLast();
            }
            deque.addLast(i);

            if (i >= k-1) {
                res[i-k+1] = nums[deque.peekFirst()];
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/sort-array-by-parity/submission-2.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        even, odd = [], []

        for num in nums:
            if num &1:
                odd.append(num)
            else:
                even.append(num)
        
        index = 0
        for e in even:
            nums[index] = e
            index +=1
        for o in odd:
            nums[index]= o
            index +=1
        
        return nums

        